module/gui/Collector.py

```diff
@@ -288,40 +860,469 @@ def headerData(self, section, orientation, role=Qt.DisplayRole):
+        
+        elif self.dnd.actionOnDrop == self.dnd.ACT.LINKORDER:
+            self.dnd.getSrcLinks()
+            if self.dnd.srcLinks:
+                if sortByOrderAttribute:
+                    self.dnd.srcLinks = sorted(self.dnd.srcLinks, key=lambda p: p[3])
+                # get the current link order
+                unsortedItems = []
+                for package in self._data:
+                    if package.id == self.dnd.srcLinksSamePackId:
+                        for child in package.children:
+                            unsortedItems.append([None, child.id])
+                # remove the selected links
+                sortedItems = list(unsortedItems)
+                for sl in self.dnd.srcLinks:
+                    sortedItems.remove([None, sl[0]])
+                # reinsert the selected links at the drop position
+                i = 0
+                while i < len(unsortedItems):
+                    if sortedItems[i][1] == self.dnd.destLinkId:
+                        for j in xrange(len(self.dnd.srcLinks)):
+                            sortedItems.insert(i + j, [None, self.dnd.srcLinks[j][0]])
+                        i += len(self.dnd.srcLinks)
+                    i += 1
+                # build a list link move commands and send them to the server
+                linkMoves = self.smallestListOfItemMoves(unsortedItems, sortedItems)
+                for move in linkMoves:
+                    self.connector.proxy.orderFile(move[0], move[1])
+                nothingDone = not linkMoves
+            else:
+                self.log.debug0("%s.slotDropEvent: LINKORDER: selection lost" % self.cname)
+        
+        elif self.dnd.actionOnDrop == self.dnd.ACT.LINKMOVE:
+            self.dnd.getSrcLinks()
+            if self.dnd.srcLinks:
+                if sortByOrderAttribute:
+                    self.dnd.srcLinks = sorted(self.dnd.srcLinks, key=lambda p: p[3])
+                fids = []
+                links = []
+                for l in self.dnd.srcLinks:
+                    fids.append(l[0])
+                    links.append(l[1])
+                # delete the links from the source package
+                self.connector.proxy.deleteFiles(fids)
+                # append (recreated) links to the destination package
+                self.connector.proxy.addFiles(self.dnd.destPackId, links)
+                nothingDone = not fids
+            else:
+                self.log.debug0("%s.slotDropEvent: LINKMOVE: selection lost" % self.cname)
+        
+        else:
+            raise ValueError("%s: Unknown drop action" % self.cname)
+        
+        if nothingDone:
+            self.view.setEnabled(self.view.corePermissions["LIST"])
+            self.view.setFocus(Qt.OtherFocusReason)
+        else:
+            self.view.clearSelection()
+            self.view.setCurrentIndex(QModelIndex())
+    
+    def advancedSelect(self, pattern, syntax, cs, deselect, selectLinks):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [CollectorModel.advancedSelect is too complex (30)](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)